Mlem/Logic/Participating/Post Comment.swift

```diff
@@ -8,120 +8,53 @@
 import Foundation
 import SwiftUI
 
-internal enum CommentPostingFailure: Error
-{
-    case couldNotPost, receivedInvalidServerResponse, coundNotParseUpdatedComments
-}
-
 @MainActor
-func postComment(to post: Post, commentContents: String, commentTracker: CommentTracker, account: SavedAccount, appState: AppState) async throws
-{
-    do
-    {
-        let commentPostingCommandResult: String = try await sendPostCommand(appState: appState, account: account, endpoint: "comment", arguments: [
-            "content": commentContents,
-            "post_id": post.id
-        ])
-
-        print("Successfuly posted comment: \(commentPostingCommandResult)")
-
-        if !commentPostingCommandResult.contains("\"error\"")
-        {
-            do
-            {
-                let postedComment: Comment = try await parseComments(commentResponse: commentPostingCommandResult, instanceLink: account.instanceLink).first!
-                
-                withAnimation(.interactiveSpring(response: 0.4, dampingFraction: 1, blendDuration: 0.4))
-                {
-                    commentTracker.comments.prepend(postedComment)
-                }
-            }
-            catch let commentParsingError
-            {
-                
-                appState.alertTitle = "Couldn't read updated comment"
-                appState.alertMessage = "Refresh comments to see your new comment."
-                appState.isShowingAlert.toggle()
-                
-                print("Failed while parsing updated comment: \(commentParsingError)")
-            }
-        }
-        else
-        {
-            print("Received error from server")
-
-            throw CommentPostingFailure.receivedInvalidServerResponse
-        }
-    }
-    catch let commentPostingError
-    {
-        print("Failed while posting comment: \(commentPostingError)")
-
-        throw CommentPostingFailure.couldNotPost
+func postComment(
+    to post: APIPostView,
```

Comment:
Why not just a post ID? (Same counts for a lot of these spots btw)

Maybe this is me being too nit picky, and this can be addressed later, but I'm noticing it here.

Reply:
While refactoring I was trying not to adjust the existing methods too much, so previously this accepted the `Post` type, I just moved it to the new model... with the changes included in the refactoring quite a few of these arguments can be simplified and in other cases I think whole methods can be dropped as they mostly just take the arguments and build the request, which could now be done in-line...

To avoid (even more) scope creep I'd suggest we address this additional tidying up as a separate PR?

Reply:
Sounds good to me, was just commenting on the oddness